Modules/Misc/GameMenuButton.lua

```diff
@@ -108,6 +108,44 @@ function M:GameMenuButton()
         collections.achievs:SetPoint("TOPLEFT", collections.pets, "BOTTOMLEFT", 0, -12)
         collections.achievs:SetFont(primaryFont, F.FontSizeScaled(16), "OUTLINE")
         collections.achievs:SetTextColor(1, 1, 1, 1)
+
+        -- Mythic+ Keystone
+        if WOW_PROJECT_ID == WOW_PROJECT_MAINLINE then
+          local keystoneFrame = CreateFrame("Frame")
+          keystoneFrame:RegisterEvent("BAG_UPDATE")
+          keystoneFrame:RegisterEvent("CHALLENGE_MODE_COMPLETED")
+          keystoneFrame:RegisterEvent("CHALLENGE_MODE_RESET")
+          keystoneFrame:RegisterEvent("CHALLENGE_MODE_START")
+          keystoneFrame:RegisterEvent("PLAYER_ENTERING_WORLD")
```

Comment:
I don't think it's necessary to register all these events for displaying a single string. The whole string display logic can be moved further down below to the `OnShow` hook and simply update the keystone each time menu is opened.